diff -Naur Csdp-6.1.1/lib/initparams.c Csdp-6.1.1-global_param.patch/lib/initparams.c
--- lib/initparams.c	2006-11-02 07:54:20.000000000 +0100
+++ lib/initparams.c	2013-02-18 09:46:02.963423012 +0100
@@ -12,6 +12,11 @@
   FILE *paramfile;
 
   paramfile=fopen("param.csdp","r");
+  /* Allow the use of global param.csdp in /usr/local/share/csdp/ */
+  if (paramfile == NULL) {
+    paramfile=fopen("/usr/local/share/csdp/param.csdp","r");
+  };
+
   if (paramfile == NULL)
     {
       params->axtol=1.0e-8;
