diff -Naur Csdp-6.1.1/include/declarations.h Csdp-6.1.1_osdp/include/declarations.h
--- Csdp-6.1.1/include/declarations.h	2009-12-02 22:28:39.000000000 +0100
+++ Csdp-6.1.1_osdp/include/declarations.h	2015-11-24 08:53:56.000000000 +0100
@@ -189,6 +189,12 @@
 	     struct blockmatrix *pX, double **py, struct blockmatrix *pZ,
 	     double *ppobj, double *pdobj);
 
+int easy_sdp_params(int n, int k, struct blockmatrix C, double *a, 
+                    struct constraintmatrix *constraints, double constant_offset,
+                    struct blockmatrix *pX, double **py, struct blockmatrix *pZ,
+                    double *ppobj, double *pdobj,
+                    struct paramstruc *params, int printlevel);
+
 void tweakgap(int n, int k, double *a, struct constraintmatrix *constraints,
 	      double gap, struct blockmatrix Z, struct blockmatrix dZ, 
 	      double *y, double *dy, struct blockmatrix work1, 
diff -Naur Csdp-6.1.1/lib/easysdp.c Csdp-6.1.1_osdp/lib/easysdp.c
--- Csdp-6.1.1/lib/easysdp.c	2006-11-02 07:54:20.000000000 +0100
+++ Csdp-6.1.1_osdp/lib/easysdp.c	2015-11-24 08:51:14.000000000 +0100
@@ -12,7 +12,7 @@
 #include <math.h>
 #include "declarations.h"
 
-int easy_sdp(n,k,C,a,constraints,constant_offset,pX,py,pZ,ppobj,pdobj)
+int easy_sdp_params(n,k,C,a,constraints,constant_offset,pX,py,pZ,ppobj,pdobj,params,printlevel)
      int n;
      int k;
      struct blockmatrix C;
@@ -24,10 +24,11 @@
      struct blockmatrix *pZ;
      double *ppobj;
      double *pdobj;
+     struct paramstruc *params;
+     int printlevel;
 {
   int ret;
   struct constraintmatrix fill;
-  struct paramstruc params;
   struct blockmatrix work1;
   struct blockmatrix work2;
   struct blockmatrix work3;
@@ -53,7 +54,7 @@
   double *dy1;
   double *rhs;
   double *besty;
-  int printlevel;
+  int free_params=0;
   int ldam;
   struct sparseblock **byblocks;
   struct sparseblock *ptr;
@@ -71,7 +72,17 @@
     *  Initialize the parameters.
     */
 
-   initparams(&params,&printlevel);
+  if (params == NULL) 
+    {
+      params=(struct paramstruc *)malloc(sizeof(struct paramstruc));
+      if (params == NULL)
+        {
+          printf("Storage Allocation Failed!\n");
+          exit(10);
+        };
+      initparams(params,printlevel < 0 ? &printlevel : &free_params);
+      free_params=1;
+    };
 
   /*
    *  Allocate working storage
@@ -436,7 +447,7 @@
 	   cholxinv,cholzinv,ppobj,pdobj,work1,work2,work3,workvec1,
 	   workvec2,workvec3,workvec4,workvec5,workvec6,workvec7,workvec8,
 	   diagO,bestx,besty,bestz,Zi,O,rhs,dZ,dX,dy,dy1,Fp,
-	   printlevel,params);
+	   printlevel,*params);
 
    if (printlevel >= 1)
      {
@@ -552,10 +563,28 @@
    * Finally, free the constraints array.
    */
 
+   if (free_params) free(params);
+
    return(ret);
 
 }
 
+int easy_sdp(n,k,C,a,constraints,constant_offset,pX,py,pZ,ppobj,pdobj)
+     int n;
+     int k;
+     struct blockmatrix C;
+     double *a;
+     struct constraintmatrix *constraints;
+     double constant_offset;
+     struct blockmatrix *pX;
+     double **py;
+     struct blockmatrix *pZ;
+     double *ppobj;
+     double *pdobj;
+{
+    return easy_sdp_params(n,k,C,a,constraints,constant_offset,pX,py,pZ,ppobj,pdobj,NULL,-1);
+}
+
 int structnnz(n,k,C,constraints)
      int n;
      int k;
diff -Naur Csdp-6.1.1/lib/Makefile Csdp-6.1.1_osdp/lib/Makefile
--- Csdp-6.1.1/lib/Makefile	2009-11-22 00:19:48.000000000 +0100
+++ Csdp-6.1.1_osdp/lib/Makefile	2014-12-18 14:50:32.000000000 +0100
@@ -24,7 +24,7 @@
 #
 # The default settings for gcc:
 #
-CFLAGS=-O3 -ansi -Wall -DNOSHORTS -DUSEGETTIME -I../include
+CFLAGS=-O3 -ansi -Wall -DNOSHORTS -DUSEGETTIME -fPIC -I../include
 #
 # Notes on CFLAGS.
 #
diff -Naur Csdp-6.1.1/Makefile Csdp-6.1.1_osdp/Makefile
--- Csdp-6.1.1/Makefile	2006-11-03 20:29:50.000000000 +0100
+++ Csdp-6.1.1_osdp/Makefile	2015-11-24 09:28:57.000000000 +0100
@@ -7,6 +7,8 @@
 #
 # Generic. On most systems, this should handle everything.
 #
+PREFIX=/usr/local
+
 all:
 	cd lib; make libsdp.a
 	cd solver; make csdp
@@ -24,12 +26,26 @@
 # Install the executables in /usr/local/bin.
 #
 
-install:
-	cp -f solver/csdp /usr/local/bin
-	cp -f theta/theta /usr/local/bin
-	cp -f theta/graphtoprob /usr/local/bin
-	cp -f theta/complement /usr/local/bin
-	cp -f theta/rand_graph /usr/local/bin
+install-bin: all
+	mkdir -p ${PREFIX}/bin
+	cp -f solver/csdp ${PREFIX}/bin
+	cp -f theta/theta ${PREFIX}/bin
+	cp -f theta/graphtoprob ${PREFIX}/bin
+	cp -f theta/complement ${PREFIX}/bin
+	cp -f theta/rand_graph ${PREFIX}/bin
+
+#
+# Install the libs and include in ${PREFIX}/{lib, include}.
+#
+
+install-lib: all
+	mkdir -p ${PREFIX}/lib/csdp
+	cp -f lib/*.o ${PREFIX}/lib/csdp
+	cp -f lib/*.a ${PREFIX}/lib/csdp
+	mkdir -p ${PREFIX}/include/csdp
+	cp -f include/*.h ${PREFIX}/include/csdp
+
+install: install-bin install-lib
 
 #
 # Clean out all of the directories.
